Add Docker Compose for Development Environment

fcancela opened the pull request:
This PR introduces a Docker Compose-based configuration to facilitate and standardize the development environment. By simplifying setup and ensuring consistency across workflows, it aims to lower the entry barrier for new contributors.

**Key Changes**:
1. **Docker Compose Configuration**:
   - Added `docker-compose.dev.yml` to define services such as:
     - Application backend.
     - Database.
     - Supporting dependencies.
   - Enables a seamless local setup with a single command.

2. **Environment Variables**:
   - Introduced `config.env.development` as the template for development-specific environment variables.
   - Ensures flexibility for contributors without exposing sensitive data.

3. **AWS Bucket Initialization**:
   - Added a `docker/aws` folder containing scripts to initialize S3 buckets for LocalStack.
   - Note: The LocalStack community version does not support persistence, so S3 buckets will be deleted each time `docker-compose` is stopped.

4. **Documentation Update**:
   - Updated the README with step-by-step instructions for setting up and running the development environment using Docker Compose.

5. **Updated Guide**:
  - Updated Transcode `docs/guide/video-processing.md` seemed to have outdated fields.

Refer to the README for detailed setup and testing instructions.

matvp91 commented:
Nice work! I've got a few remarks though:

- This worries me a bit: "The LocalStack community version does not support persistence, so S3 buckets will be deleted each time docker-compose is stopped.", as Superstreamer will register an asset in the DB but it won't be available when docker images are spun up again. I found this: https://github.com/GREsau/localstack-persist, that might be a resolution?

- Can you remove the added comments in the README.md? I haven't figured out yet how I'd like to structure the README but I'd like to keep it short and direct people to the docs. Feel free to add another section in the docs (https://superstreamer.xyz/guide/getting-started.html#local-builds) under "Local build" though!

fcancela commented:
Great findings! 

I'm now using [localstack-persist](https://github.com/GREsau/localstack-persist).
I also moved the quick dev setup guide to the [Getting Started local builds section](https://superstreamer.xyz/guide/getting-started.html#local-builds).

I've added an addition on how to handle environment variables for the dev environment so the users don't need to set them up manually (NODE_ENV) and using [bun config file conventions](https://bun.sh/docs/runtime/env#setting-environment-variables). Let me know if you find this useful, if not, I can roll back those changes.

matvp91 commented on docs/guide/getting-started.md:
````diff
@@ -233,4 +234,55 @@ We've already covered how to build Superstreamer locally, and we've also made it
+We have also created a `docker-compose-dev.yml` so you can setup your development environment faster and start getting hands on!
+
+```shell
+# We have prebuilt developmnet containers, see docker/docker-compose-dev.yml
````
typo, developmnet -> development

matvp91 commented on docs/guide/getting-started.md:
````diff
@@ -233,4 +234,55 @@ We've already covered how to build Superstreamer locally, and we've also made it
+docker-compose -f docker-compose-dev.yml up
+```
+
+You can use this `config.env.development` for a quick setup:
````
Since `config.env.development` as file name would be ignored, it's probably best to mention that people have to paste these env variables in `config.env` instead.

fcancela replied:
I added a clarification:

> "You can create a file named `config.env.development` for a quick setup. Here is a sample that should work out of the box if default configuration is used:"

matvp91 commented on docs/guide/getting-started.md:
````diff
@@ -233,4 +234,55 @@ We've already covered how to build Superstreamer locally, and we've also made it
+docker-compose -f docker-compose-dev.yml up
+```
+
+You can create a file named `config.env.development` for a quick setup. Here is a sample that should work out of the box if default configuration is used:
````
shouldn't config.env.development be renamed to config.env?
